### PR TITLE
Fix local packages linkage in docker prod build

### PR DESCRIPTION
**Pull request by clemsos**
# Description

Use local `link:` protocol to prevent yarn from fetching local packages and throwing 

```
#42 1.509 error An unexpected error occurred: "https://registry.yarnpkg.com/@unlock-protocol%2fnetworks: Not found".
```

https://app.circleci.com/pipelines/github/unlock-protocol/unlock/18145/workflows/3d49ce55-bb70-4cf9-8574-914aa0f48a5b/jobs/252263

# Issues

Relates to https://github.com/yarnpkg/yarn/issues/2611 and https://github.com/yarnpkg/yarn/issues/5083

# Checklist:

- [ ] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -125,15 +125,19 @@ WORKDIR /app
 COPY --from=build --chown=node /home/unlock/$BUILD_DIR/package.json package.json
 COPY --from=build --chown=node /home/unlock/$BUILD_DIR/yarn.lock yarn.lock
 
-# delete dev deps to prevent yarn from fetching them
+# delete dev deps (prevents yarn from fetching them)
+# add link: protocol workaround for yarn local links to work
 RUN apk add --no-cache --virtual .build-deps coreutils jq  \
-    && jq 'del(.devDependencies)' package.json > package.json.tmp \
+    && jq 'del(.devDependencies) | (.dependencies |= (with_entries(if .key == "@unlock-protocol/networks" then .key = "link:./packages/networks" else . end)))' package.json > package.json.tmp \
```
Comment (julien51) [inline Dockerfile:131]: Tu penses que ca peut valoir le coup de faire un truc plus generique car j'ai peur qu'on ait d'autres packages du meme genre?
Sinon, je me disais que les packages internes on a peut etre pas besoin de les prefix abec `@unlock-protocol`, wdyt?

Comment (clemsos) [reply Dockerfile:131]: pour le prefixe, je pense que c'est bien que ce soit consistant et c'est utile pour yarn workspace. Par contre on peut ajouter `private: true` dans le package pour être sûr de ne pas le publier.

OK pour faire un truc plus générique i.e. remplacer par une regexp. Je me disais juste qu'on ferait ça le moment venu :P

Comment (julien51) [reply Dockerfile:131]: HA! noted :) Let's do it now because I worry we'll forget about this in 3 months and be confused! Thanks!

Comment (clemsos) [reply Dockerfile:131]: If we decide to write a regex to replace `@unlock-protocol/` by `link:./packages` then we will run across the problem that `@unlock-protocol/unlock-js` or `@unlock-protocol/contract-abi-v-x` are not there. Of course the best will be for yarn to solve this

**Review by julien51 — APPROVED** [review]

**Pull request merged**